consensus/propagation/reactor_test.go

```diff
@@ -431,6 +432,131 @@ func TestPropagationSmokeTest(t *testing.T) {
+			Src:       reactor2Peer,
+			Message:   &propagation.CompactBlock{BpHash: cmtrand.Bytes(2)},
+			ChannelID: byte(0x05),
+		}
+		reactor1.ReceiveEnvelope(invalidEnvelope)
+		assert.Nil(t, reactor1.getPeer(reactor2.self))
+	})
+	t.Run("invalid compact block: incorrect leaf hash", func(t *testing.T) {
+		// TODO implement this test case
+	})
+	t.Run("have part for unknown proposal", func(t *testing.T) {
+		reactors, _ := testBlockPropReactors(2, cfg.DefaultP2PConfig())
+		reactor1 := reactors[0]
+		reactor2 := reactors[1]
+		require.NotNil(t, reactor1.getPeer(reactor2.self))
+		// send haves before receiving a proposal
+		reactor1.handleHaves(reactor2.self, &proptypes.HaveParts{
+			Height: 1,
+			Round:  2,
+		})
+		assert.Nil(t, reactor1.getPeer(reactor2.self))
+	})
+	t.Run("want part for unknown proposal", func(t *testing.T) {
+		reactors, _ := testBlockPropReactors(2, cfg.DefaultP2PConfig())
+		reactor1 := reactors[0]
+		reactor2 := reactors[1]
+		require.NotNil(t, reactor1.getPeer(reactor2.self))
+		// send wants before receiving a proposal
+		reactor1.handleWants(reactor2.self, &proptypes.WantParts{
+			Height: 1,
+			Round:  2,
+		})
+		assert.Nil(t, reactor1.getPeer(reactor2.self))
+	})
+	t.Run("recovery part for unknown proposal", func(t *testing.T) {
+		reactors, _ := testBlockPropReactors(2, cfg.DefaultP2PConfig())
+		reactor1 := reactors[0]
+		reactor2 := reactors[1]
+		require.NotNil(t, reactor1.getPeer(reactor2.self))
+		// send wants before receiving a proposal
+		reactor1.handleRecoveryPart(reactor2.self, &proptypes.RecoveryPart{
+			Height: 1,
+			Round:  2,
+		})
+		assert.Nil(t, reactor1.getPeer(reactor2.self))
+	})
+}
+
+// testCompactBlock returns a test compact block with the corresponding orignal part set,
+// parity partset, and proofs.
+func testCompactBlock(t *testing.T, height int64, round int32) (*proptypes.CompactBlock, *types.PartSet, *types.PartSet, []*merkle.Proof) {
+	ps := types.NewPartSetFromData(cmtrand.Bytes(1000), types.BlockPartSizeBytes)
+	pse, lastLen, err := types.Encode(ps, types.BlockPartSizeBytes)
+	require.NoError(t, err)
+	psh := ps.Header()
+	pseh := pse.Header()
+
+	hashes := extractHashes(ps, pse)
+	proofs := extractProofs(ps, pse)
+
+	baseCompactBlock := &proptypes.CompactBlock{
+		BpHash:    pseh.Hash,
+		Signature: cmtrand.Bytes(64),
+		LastLen:   uint32(lastLen),
+		Blobs: []proptypes.TxMetaData{
+			{Hash: cmtrand.Bytes(32)},
+			{Hash: cmtrand.Bytes(32)},
+		},
+		PartsHashes: hashes,
+	}
+
+	// adding the proposal manually so the haves/wants and recovery
+	// parts are not rejected.
+	p := types.Proposal{
+		BlockID: types.BlockID{
+			Hash:          cmtrand.Bytes(32),
+			PartSetHeader: psh,
+		},
+		Height: height,
+		Round:  round,
+	}
+	baseCompactBlock.Proposal = p
+
+	return baseCompactBlock, ps, pse, proofs
+}
+
+// TestHugeBlock doesn't have a success or failure condition yet, although one could be added. It is very useful for debugging however
+func TestHugeBlock(t *testing.T) {
```

Comment:
@evan-forbes this can be deleted and just keep the smoke test, right?

Reply:
yeah just keep the smoke test

why is it being added here? did merging the new branch do something weird?